src/clj_telnet/core.clj

```diff
@@ -31,6 +34,26 @@
   ([^String server-ip]
    (get-telnet server-ip 23)))
 
+(defn  print-c-debug [c]
+  (when *debug* (print (char c))))
+
+(defn print-data-debug [data]
+  (when *debug* (print data)))
+
+(defn- read-a-char
+  [in]
+  (try
+    (let [c (.read in)]
+      (print-c-debug c)
+      c)
+    (catch SocketTimeoutException ste (println "Read error") nil)))
+
+(defn- write-data
```

Comment:
What if write-data and read-char were ^:dynamic instead of switching on *debug*?
Alternatively this library would benefit from proper logging like org.clojure/tools.logging